### PR TITLE
Update docs for rspec-rails 4

### PR DESCRIPTION
**Pull request by JonRowe**
When publishing the cukes I found this, there might be more...

### CODE REVIEW

File: features/README.md [hunk 1]
```diff
@@ -4,9 +4,8 @@ routing.
 
 ## Rails
 
-rspec-rails 3 supports Rails 3.x and 4.x. For earlier versions of Rails, you
-need [rspec-rails 1](https://github.com/dchelimsky/rspec-rails). For Rails 5.x,
-make sure you are using rspec-rails 3.5.0 or newer.
+rspec-rails 4 supports Rails 5.x and 6.x. For earlier versions of Rails, you
```
Comment (pirj) [inline features/README.md]: Readme as well mentions decade-old Rails https://github.com/rspec/rspec-rails/blame/master/README.md#L11

Comment (pirj) [reply features/README.md]: Also in README:
```
**IMPORTANT** Note that rspec-rails 4.0 is still a release candidate
```

Comment (pirj) [reply features/README.md]:
```
     # Note that rspec-rails 4.0 is still a pre release
     gem 'rspec-rails', '~> 4.0.0.rc1'
```

Comment (JonRowe) [reply features/README.md]: You might want to update your master, thats gone :joy:

Comment (pirj) [reply features/README.md]: features/upgrade/README.md needs some cleanup:
```
With the release of Rails 4, new APIs have been exposed on
`ActiveRecord::Migration`. This allows RSpec to take advantage of these new
standard migration checks, mirroring behavior across the board.
```

Comment (pirj) [reply features/README.md]: lib/generators/rspec/integration/integration_generator.rb:
```
      # Add a deprecation for this class, before rspec-rails 4, to use the
      # `RequestGenerator` instead
```
drop the file completely?

Comment (pirj) [reply features/README.md]: features/upgrade/README.md might need some additions, but I hope user-submitted ones. Frankly I lost track of what has been added and what needs to be tweaked before updating `rspec-rails` on a project.

We should say goodbye to `from_1x_to_2x.md`.

Comment (pirj) [reply features/README.md]: That's all I could find.

**Comment by benoittgt** [comment]
Also the description of the github repo could be changed: "RSpec for Rails-3+"

File: features/README.md (hunk 1 above)
Comment (JonRowe) [reply features/README.md]: @pirj would you mind tackling those in another PR? I'm going to merge this to get the build green.

Comment (pirj) [reply features/README.md]: I'll take care of it 👍

**Pull request merged**